frontend/src/context/layers/composite_data.ts

```diff
@@ -12,26 +13,30 @@ export interface CompositeLayerData extends FeatureCollection {}
 export const fetchCompositeLayerData: LazyLoader<CompositeLayerProps> =
   () =>
   async (params: LayerDataParams<CompositeLayerProps>, { dispatch }) => {
-    // to complete later with new endpoint for composite chart
-
     const { layer, date } = params;
-    const endDate = (date ? new Date(date) : new Date())
-      .toISOString()
-      .split('T')[0];
-    const { baseUrl, inputLayers, startDate } = layer;
+    const startDate = date ? new Date(date) : new Date();
+    const endDate = new Date(startDate);
+    endDate.setMonth(endDate.getMonth() + 1);
+
+    const {
+      baseUrl,
+      inputLayers,
+      startDate: areaStartDate,
+      endDate: areaEndDate,
+    } = layer;
     const { boundingBox } = appConfig.map;
 
     // docs: https://hip-service.ovio.org/docs#/default/run_q_multi_geojson_q_multi_geojson_post
     const body = {
-      begin: startDate,
-      end: endDate,
+      begin: getFormattedDate(startDate, 'default'),
+      end: getFormattedDate(endDate, 'default'),
       area: {
         min_lon: boundingBox[0],
         min_lat: boundingBox[1],
         max_lon: boundingBox[2],
         max_lat: boundingBox[3],
-        start_date: '2002-01-01',
-        end_date: endDate,
+        start_date: areaStartDate ?? '2002-01-01',
+        end_date: areaEndDate ?? '2021-07-31',
```

Comment:
Not sure I fully understand so bear with me. But I would be expecting this to take the max of a default and endDate, so max('2021-07-31', endDate).
But maybe we actually want the baseline to be stable over time so 🤷

Reply:
I _think_ the max of 2021-07-31 would be mainly relevant for CDI and not necessarily all future composite layers (but I could totally be wrong about that). 

In general, I'd say we should limit the role of hardcoded dates and prefer for this type of data to come from the layer.json. If the layer.json gives us date to use, prism should trust it and not overrule it. If there are issues with the date provided, we fix the layer.json. I'd prefer to not have 2021-07-31 in here at all and only kept it to reduce risk of regressions.

But take this all with a grain of salt, definitely still learning my way around this project 😅